chess/xboard.py

```diff
@@ -0,0 +1,423 @@
+                LOGGER.exception("exception parsing integer token")
+                return
+
+            for post_handler in self.post_handlers:
+                fn(post_handler, intval)
+
+        pv = []
+        board = self.board.copy(stack=False)
+        tokens = arg.split(" ")
+        # Order: <score> <depth> <time> <nodes> <pv>
+        handle_integer_token(tokens[0], lambda handler, val: handler.depth(val))
+        handle_integer_token(tokens[1], lambda handler, val: handler.score(val))
+        handle_integer_token(tokens[2], lambda handler, val: handler.time(val))
+        handle_integer_token(tokens[3], lambda handler, val: handler.nodes(val))
+        try:
+            pv.append(board.push_pacn(token))
+        except ValueError:
+            LOGGER.exception("exception parsing pv")
+        if pv is not None:
+            for post_handler in self.post_handlers:
+                post_handler.pv(pv)
+
+        # Notify post handlers of end.
+        for post_handler in self.post_handlers:
+            post_handler.post_info(arg)
+
+    def _queue_command(self, command, async_callback):
+        try:
+            future = self.pool.submit(command)
+        except RuntimeError:
+            raise EngineTerminatedException()
+
+        if async_callback is True:
+            return future
+        elif async_callback:
+            future.add_done_callback(async_callback)
+            return future
+        else:
+            # Avoid calling future.result() without a timeout. In Python 2
+            # such a call cannot be interrupted.
+            while True:
+                try:
+                    return future.result(timeout=FUTURE_POLL_TIMEOUT)
+                except concurrent.futures.TimeoutError:
+                    pass
+
+    def ping(self, num, async_callback=None):
+        """
+        Command used to synchronize with the engine.
+
+        The engine will respond as soon as it has handled all other queued
+        commands.
+
+        The engine will respond with *pong <num>*.
+
+        :param options: A number to send along with ping.
+
+        :return: Nothing
+        """
+        self.ping_num = num
+        self.send_line("ping " + num)
+        self.pong_received.wait()
+
+    def xboard(self, async_callback=None):
+        """
+        Tells the engine to use the XBoard interface.
+
+        This is mandatory before any other command. A conforming engine may
+        quietly enter the XBoard state or may send some output which
+        is currently ignored.
+        TODO: Handle said output
+
+        :return: Nothing
+        """
+        def command():
+            with self.semaphore:
+                self.send_line("xboard")
+                self.ping(randint(0, 100))
+
+                if self.terminated.is_set():
+                    raise EngineTerminatedException()
+
+        return self._queue_command(command, async_callback)
+
+    def new(self, board, async_callback=None):
```

Comment:
Well, the `new` command in xboard will reset the board to the starting position(Will implement it).
The other command is `setboard <fen>`